Secret handling

Khan-Ramsha opened the pull request:
Fix for #57 

Changes made:
1) In `middleware.py` :  Replaced the hardcoded secret value with SECRET_KEY loaded from the environment.
2) In `settings.py` : Updated FromOriginMatchHeader to use SECRET_KEY as the value for secret.
3) Create .env file which contains SECRET_KEY = "uNz9B$2F7&J@aKqP1!X"

Additional changes, This PR also includes updates from [PR](https://github.com/myhealthconnectsociety/project-healthcare/pull/115), which updates the README file and includes project documentation.

Please review the changes, and let me know if any further changes are needed. Thankyou!

## Summary by Sourcery

Improve secret handling by loading SECRET_KEY from the environment and update project documentation.

Bug Fixes:
- Replace hardcoded secret value with SECRET_KEY loaded from the environment in middleware and settings.

Documentation:
- Update README with new project details, installation steps, and deployment instructions.
- Add detailed project documentation in a new file, including product details, hardware and software requirements, and core APIs.

Khan-Ramsha committed:
Update README:added documentation,installation steps, description,vision and sequence diagram

Khan-Ramsha committed:
fix: moving documentation into repo's /docs

Khan-Ramsha committed:
fix: moving documentation into repo's /docs folder

Khan-Ramsha committed:
fix: moving the img to the main repo..

Khan-Ramsha committed:
Merge branch 'myhealthconnectsociety:main' into main

Khan-Ramsha committed:
fix: created documentation folder in the project root

Khan-Ramsha committed:
fix: redirect to project documentation

Khan-Ramsha committed:
chore: added documentation

Khan-Ramsha committed:
updated readme..

Khan-Ramsha committed:
updated readme..

Khan-Ramsha committed:
fix: renamed documentation to docs

Khan-Ramsha committed:
fixed the link to README file

Khan-Ramsha committed:
refactor: replace hardcoded secret with env var

Khan-Ramsha committed:
fix: typo

codecakes commented:
> Hey @Khan-Ramsha - I've reviewed your changes - here's some feedback:
> 
> **Overall Comments**:
> 
> * Remove the .env file containing secrets from version control. Instead, add a .env.example template file and add .env to .gitignore. Never commit actual secrets to the repository.
> * Update the documentation to explain the required environment variables and configuration steps, including the new SECRET_KEY requirement.
> 
> Here's what I looked at during the review
> Sourcery is free for open source - if you like our reviews please consider sharing them ✨
> Help me be more useful! Please click 👍 or 👎 on each comment and I'll use the feedback to improve your reviews.

@Khan-Ramsha there are few reviews here already

Khan-Ramsha committed:
updated documentation & added .env.example

Khan-Ramsha committed:
Update project_healthcare_documentation.md

Khan-Ramsha committed:
Update project_healthcare_documentation.md

Khan-Ramsha commented:
> there are few reviews here already
Yes, sir. I have updated the documentation to include details about the required environment variables and configuration steps, specifically the new SECRET_KEY requirement. Additionally, I have added a `.env.example` template file for reference. Please note that the `.env`  file itself is not tracked by Git, as it has already been added to the `.gitignore` file.

Khan-Ramsha commented:
> there are few reviews here already


Yes, sir. I have updated the documentation to include details about the required environment variables and configuration steps, specifically the new SECRET_KEY requirement. Additionally, I have added a `.env.example` template file for reference. Please note that the `.env`  file itself is not tracked by Git, as it has already been added to the `.gitignore` file.

Khan-Ramsha committed:
Merge branch 'main' into secret-handling
